src/ApiCore/GapicClientTrait.php

```diff
@@ -391,41 +409,23 @@ protected function startCall(
      *
      * @return callable
      */
-    protected function createCallStack(array $callConstructionOptions)
+    private function createCallStack(array $callConstructionOptions)
     {
-        return new RetryMiddleware(
-            new AgentHeaderMiddleware(
-                new CredentialsWrapperMiddleware(
-                    function (Call $call, array $options) {
-                        $startCallMethod = $this->transportCallMethods[$call->getCallType()];
-                        return $this->transport->$startCallMethod($call, $options);
-                    },
-                    $this->authWrapper
-                ),
-                $this->agentHeaderDescriptor
-            ),
-            $callConstructionOptions['retrySettings']
-        );
-    }
-
-    /**
-     * @param array $optionalArgs {
-     *     Optional arguments
-     *
-     *     @type array $headers [optional] key-value array containing headers
-     *     @type int $timeoutMillis [optional] the timeout in milliseconds for the call
-     *     @type array $transportOptions [optional] transport-specific call options
-     * }
-     *
-     * @return array
-     */
-    protected function configureCallOptions(array $optionalArgs)
-    {
-        return $this->pluckArray([
+        $callable = function (Call $call, array $options) {
+            $startCallMethod = $this->transportCallMethods[$call->getCallType()];
+            return $this->transport->$startCallMethod($call, $options);
+        };
+        $callable = new CredentialsWrapperMiddleware($callable, $this->credentialsWrapper);
+        $callable = new AgentHeaderMiddleware($callable, $this->agentHeaderDescriptor);
+        $callable = new RetryMiddleware($callable, $callConstructionOptions['retrySettings']);
+        $callable = new OptionsFilterMiddleware($callable, [
             'headers',
             'timeoutMillis',
             'transportOptions',
-        ], $optionalArgs);
+            'metadataCallback',
```

Comment:
Should we document this alongside `headers`, `timeoutMillis`, etc. or did you want to keep this as a "hidden" feature for now?